cmd/root.go

```diff
@@ -47,5 +48,12 @@ func init() {
 	rootCmd.AddCommand(factory.Build(cli.WhoamiCommand))
 
 	// pass a list of functions to be run before each command's Execute function
-	cobra.OnInitialize(func() { cli.HandleErr(profile.Load()) })
+	cobra.OnInitialize(func() { handleErr(profile.Load()) })
+}
+
+func handleErr(err error) {
+	if err == nil {
+		return
+	}
+	log.Fatal(err)
 }
```

Comment:
is this the final result of this function? if so - can we just handle these in line to the calling func?

Reply:
I toyed with the idea of this function "doing more", like checking for types of errors and subsequently printing more information if need be, but the more i think about it the more I think that all can be provided simply through the error string.

just for clairty: the two pieces of info on my mind that will want to _also_ be printed are any potential docs links related to the error that occurred, and any potential follow-up commands to run instead of the just-failed one.

so yeah, long story long, I think this is the final result of this function (but no regis, that's not my final answer).  right now, it's just a nil-safe process-ender, privately scoped to this `cmd/root` file and used in the 3 error-prone spots of this code.  if you'd like, more than happy to delete this helper and do the error check/log.Fatal dance in each spot above

Reply:
going to close this thread in favor of the comment [here](https://github.com/10gen/realm-cli/pull/26/files#r529842636) so we can continue it if need be